### PR TITLE
Use SDK instead of direct socket connection

### PR DESCRIPTION
**Pull request by prydin**
This will make the code a lot more resilient. 

Can you please review @vikramraman

### CODE REVIEW

File: backend/writer.go [hunk 1]
```diff
@@ -1,98 +1,69 @@
...
-	Metric    string
-	Value     float64
-	Timestamp int64
-	Source    string
-	Tags      map[string]string
-}
-
-func NewMetricWriter(host, prefix string, tags map[string]string) *MetricWriter {
+func NewMetricWriter(host string, port int, prefix string, tags map[string]string) (*MetricWriter, error) {
+	sender, err := senders.NewProxySender(&senders.ProxyConfiguration{
```
Comment (vikramraman) [inline backend/writer.go]: Could we support the direct sender option as well? Could be useful for POCs etc.

**Review by vikramraman — COMMENTED** [review]
just a minor comment. lgtm otherwise.

**Comment by vikramraman** [comment]
@prydin travis error just seems to be gofmt related. Could you just format the files?

**Comment by isugimpy** [comment]
Is it intentional that this PR stops changing underscores into dots?

**Comment by vikramraman** [comment]
Likely not intentional. There is a separate issue #16 to make that aspect configurable. But we will make sure to retain backwards compat as we address that.

**Review by vikramraman — APPROVED** [review]

**Pull request merged**